app/parse.py

```diff
@@ -8,5 +16,35 @@
     duration: str
 
 
-def get_all_courses() -> list[Course]:
-    pass
+def get_course_info_soup(course_soup: BeautifulSoup):
+    course_info_page = course_soup.select_one("a")["href"]
+    course_url = urljoin(BASE_URL, course_info_page)
+
+    course_page = requests.get(course_url).content
+
+    course_info_soup = BeautifulSoup(course_page, "html.parser")
+
+    return course_info_soup
+
+
+def parse_course_name(course_soup: BeautifulSoup) -> Course:
+    return Course(
+        name=course_soup.select_one("a")["title"],
+        short_description=get_course_info_soup(course_soup).find(
+            class_="typography_landingTextMain__Rc8BD SalarySection_aboutProfession__1VFHK"
+        ).text,
```

Comment:
The code assumes that the element with class 'typography_landingTextMain__Rc8BD SalarySection_aboutProfession__1VFHK' is always present. Consider adding a check to ensure that this element exists before accessing its text to avoid potential errors.